### PR TITLE
Ruby group capstone

### PR DESCRIPTION
**Pull request by K0ppai**
Hello Reviewers,
We've worked on the following in this pull request,

#### Logic

- Each class from the [UML class diagram](./images/catalog_of_my_things.png) created in a separate .rb file.
- All associations (1-to-many relationships) and aggregations (parent-child relationships) between classes are implemented.
- All properties visible in the diagram are defined and set up in the constructor method.
- All methods visible in the diagram implemented:
    - add_item method (in all classes that have that method)
    - can_be_archived?() in the Item class
    - move_to_archive() in the Item class
    - can_be_archived?() in the Book class, MusicAlbum class, the Game class
- Added unit tests for all implemented methods.

#### User interface

- Created a main.rb file that will serve as a console app entry point.
- Our console app, at the start, will:
    - Present the user with a list of options to perform.
    - Let users choose an option.
    - If needed, ask for parameters for the option.
    - Have a way to quit the app.
- The following options are available:
    - List all books
    - List all music albums
    - List of games
    - List all genres (e.g. 'Comedy', 'Thriller')
    - List all labels (e.g. 'Gift', 'New')
    - List all authors (e.g. 'Stephen King')
    - Add a book
    - Add a music album
    - Add a game
 - All data is preserved by saving collections in .json files.

### Database
- Created a schema.sql file with tables that will be analogical to the structure of the classes in our app.

### CODE REVIEW

File: main.rb [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+  end
+
+  def display_options(selected_option)
+    options = [
+      '1️⃣   List all books 📚',
+      '2️⃣   List all music albums 🎶 📀',
+      '3️⃣   List of games 🎮',
+      '4️⃣   List all genres  🎵 🎶',
+      '5️⃣   List all labels 🏷️  📚',
+      '6️⃣   List all authors ✍️  📖',
```
Comment (tresorsawasawa) [inline main.rb:72]: - [ ] As required for the project your application to work correctly, kindly ensure that the selected options above are working as expected. Right now, while having books, music_albums, games,... in your data base, nothing is working with selected options above

File: spec/Book/book_spec.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
+require_relative '../spec_helper'
+
```
Comment (tresorsawasawa) [inline spec/Book/book_spec.rb:2]: - [ ] As required for the project to have unit tests to all public methods in all classes should, kindly ensure that you also implement tests for the `Item` class and to the public method inside.

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @K0ppai, @sunga12, and @batoolfatima2135 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- All tests are passing :heavy_check_mark: 
- Respect associations between classes :heavy_check_mark: 
- Used primary key & foreign key mechanisms for joining tables. :100: :heavy_check_mark: 
- Preserve data in `json` files :heavy_check_mark: 
- Follow GitHub flow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by V-Blaze — APPROVED** [review]
Hi @K0ppai, @batoolfatima2135, @sunga12   

# Approved 🟢 🟢 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
giphy

## Highlights

- All rests are passing 💯
-  Meaningful commit messages👍
-  UI look great🟢 👍

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**